build-rust-crate/builder-bin.sh

```diff
@@ -59,7 +59,12 @@ configurePhase() {
 
     convertCargoToml
 
-    globalEdition="$(jq --raw-output '.package.edition // ""' "$cargoTomlJson")"
+    if [[ -z "$edition" ]]; then
+        globalEdition="$(jq --raw-output '.package.edition // ""' "$cargoTomlJson")"
+    else
+        globalEdition="$edition"
+    fi
```

Comment:
This change is related to the workspace inheritance, because crates are allowed to inherit the edition from the main workspace, and so I defined `edition` as a variable that we derived from the workspace and only check the `Cargo.toml` if that isn't defined (because we don't have access to the workspace's toml at the build script).

Reply:
> This change is related to the workspace inheritance, because crates are allowed to inherit the edition from the main workspace, and so I defined `edition` as a variable that we derived from the workspace and 

Ok, I understood. But again, all workspace inheritance related changes should be split into another PR.

> only check the `Cargo.toml` if that isn't defined

Also according the [cargo reference](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table), workspace inheritance only happens when explicitly specified `{key}.workspace = true`.

> (because we don't have access to the workspace's toml at the build script).

We can make it possible by passing additional attribute like `workspaceCargoToml`. I don't think one workspace can spread in multiple repositories, or can it?

Reply:
> Also according the [cargo reference](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table), workspace inheritance only happens when explicitly specified {key}.workspace = true.
Yes, what I meant by not defined is that that line was returning `{workspace : true}` instead of the year of the edition.

> We can make it possible by passing additional attribute like workspaceCargoToml. I don't think one workspace can spread in multiple repositories, or can it?
I think the actual problem is that you need to crawl these workspaces, because a workspace might have a nested workspace that inherits from the first one, AFAIK (I might be mistaken). Otherwise that sounds plausible.